arithmetization/src/main/java/net/consensys/linea/zktracer/types/TransactionProcessingMetadata.java

```diff
@@ -175,14 +176,16 @@ public void setPreFinalisationValues(
   }
 
   public void completeLineaTransaction(
-      Hub hub, final boolean statusCode, final List<Log> logs, final Set<Address> selfDestructs) {
+      Hub hub,
+      WorldView world,
+      final boolean statusCode,
+      final List<Log> logs,
+      final Set<Address> selfDestructs) {
     this.statusCode = statusCode;
     hubStampTransactionEnd = hub.stamp();
     this.logs = logs;
     for (Address address : selfDestructs) {
-      destructedAccountsSnapshot.add(
-          AccountSnapshot.fromAddress(
-              address, true, hub.deploymentNumberOf(address), hub.deploymentStatusOf(address)));
+      destructedAccountsSnapshot.add(AccountSnapshot.canonical(hub, world, address));
```

Comment:
This was a very stupid oversight / temporary solution to a problem that only became noticeable with account consistency on. The `fromAddress` assigns empty code, zero balance, zero nonce etc. to the `AccountSnapshot` it is creating. Unsurprisingly this broke account consistency constraints.